packages/blade/src/components/Typography/Text/Text.tsx

```diff
@@ -19,7 +18,7 @@ type TextCommonProps = {
   contrast?: ColorContrastTypes;
   truncateAfterLines?: number;
   children: React.ReactNode;
-  weight?: keyof Theme['typography']['fonts']['weight'];
+  weight?: Extract<BaseTextProps['fontWeight'], 'regular' | 'medium' | 'semibold'>;
```

Comment:
why not extract from typography itself?

Reply:
Text extends on top of BaseText so taking from that is more stable no? image we add more typography font weights but not add it in BaseText. Since we pass down weight from Text to BaseText, there might be something that shows up in type of Text but is not handled on BaseText.